debezium-connector-postgres/src/main/java/io/debezium/connector/postgresql/connection/PostgresDefaultValueConverter.java

```diff
@@ -0,0 +1,177 @@
+            if (schemaBuilder == null) {
+                return defaultValue;
+            }
+            final Schema schema = schemaBuilder.build();
+            // In order to get the valueConverter for this column, we have to create a field;
+            // The index value -1 in the field will never used when converting default value;
+            // So we can set any number here;
+            final Field field = new Field(column.name(), -1, schema);
+            final ValueConverter valueConverter = valueConverters.converter(column, field);
+            Object result = valueConverter.convert(defaultValue);
+            if ((result instanceof BigDecimal) && column.scale().isPresent() && column.scale().get() > ((BigDecimal) result).scale()) {
+                // Note that as the scale is increased only, the rounding is more cosmetic.
+                result = ((BigDecimal) result).setScale(column.scale().get(), RoundingMode.HALF_EVEN);
+            }
+            return result;
+        }
+        return defaultValue;
+    }
+
+    private static Map<String, DefaultValueMapper> createDefaultValueMappers(TimestampUtils timestampUtils) {
+        final Map<String, DefaultValueMapper> result = new HashMap<>();
+
+        result.put("bit", v -> {
+            String defaultValue = extractDefault(v, "00"); // if default is generated, assume length > 1
+            if (defaultValue.length() == 1) {
+                // treat as a bool
+                return "1".equals(defaultValue);
+            }
+            return defaultValue;
+        }); // Sample values: `B'1'::"bit"`, `B'11'::"bit"`
+        result.put("varbit", v -> extractDefault(v, "0")); // Sample value: B'110'::"bit"
+
+        result.put("bool", v -> Boolean.parseBoolean(extractDefault(v, "false"))); // Sample value: true
+
+        result.put("bpchar", v -> extractDefault(v, "")); // Sample value: 'abcd'::bpchar
+        result.put("varchar", v -> extractDefault(v, "")); // Sample value: `abcde'::character varying
+        result.put("text", v -> extractDefault(v, "")); // Sample value: 'asdf'::text
+
+        result.put("numeric", v -> new BigDecimal(extractDefault(v, "0.0"))); // Sample value: 12345.67891
+        result.put("float4", v -> Float.parseFloat(extractDefault(v, "0.0"))); // Sample value: 1.234
+        result.put("float8", v -> Double.parseDouble(extractDefault(v, "0.0"))); // Sample values: `1.234`, `'12345678901234567890'::numeric`
+        result.put("int2", v -> Short.parseShort(extractDefault(v, "0"))); // Sample value: 32767
+        result.put("int4", v -> Integer.parseInt(extractDefault(v, "0"))); // Sample value: 123
+        result.put("serial", v -> Integer.parseInt(extractDefault(v, "0")));
+        result.put("int8", v -> Long.parseLong(extractDefault(v, "0"))); // Sample values: `123`, `'9223372036854775807'::bigint`
+        result.put("bigserial", v -> Long.parseLong(extractDefault(v, "0")));
```

Comment:
I've included `SERIAL` types as these are inherently non-null but may still be added to an existing schema. These types have a default value of `nextval('sequence_name')` so will always propagate '0'

Strangely, `SMALLSERIAL` gets a type of `int2` here, while the others have their own type. Not sure if there is a reason for that inconsistency or if it is possibly a bug